### PR TITLE
BO : Right side block of attribute and attribute groups in combination tab not appears.

### PR DESCRIPTION
**Pull request by dineshbadrukhiya**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | In multi-shop and multi-language, Right side block of attribute and attribute groups in combination tab not appears.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/10377
| How to test?  | Check the back-office right side block in combination tab.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10392)

### CODE REVIEW

File: src/PrestaShopBundle/Controller/Admin/ProductController.php [hunk 1]
```diff
@@ -543,7 +543,7 @@ public function formAction($id, Request $request)
         }
 
         $doctrine = $this->getDoctrine()->getManager();
-        $attributeGroups = $doctrine->getRepository('PrestaShopBundle:Attribute')->findByLangAndShop(1, 1);
+        $attributeGroups = $doctrine->getRepository('PrestaShopBundle:Attribute')->findByLangAndShop((int)$languages[0]['id_lang'], (int)['shop_id']);
```
Comment (MathiasReker) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: Should be:
`$attributeGroups = $doctrine->getRepository('PrestaShopBundle:Attribute')->findByLangAndShop((int) $languages[0]['id_lang'], (int) ['shop_id']);
`

**Review by MathiasReker — COMMENTED** [review]
code standard

**Review by Akashvastarpara — DISMISSED** [review]

File: src/PrestaShopBundle/Controller/Admin/ProductController.php [hunk 2]
```diff
@@ -543,7 +543,7 @@ public function formAction($id, Request $request)
         }
 
         $doctrine = $this->getDoctrine()->getManager();
-        $attributeGroups = $doctrine->getRepository('PrestaShopBundle:Attribute')->findByLangAndShop(1, 1);
+        $attributeGroups = $doctrine->getRepository('PrestaShopBundle:Attribute')->findByLangAndShop((int) $languages[0]['id_lang'], (int) ['shop_id']);
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php]: You must make sure `$languages` has `[0]` and `['shop_id']` isn't an id. 
You can do:
```php
$language = empty($languages[0]) ? ['id_lang' => 1, 'shop_id' => 1] ? $language[0];
$attributeGroups = $doctrine->getRepository('PrestaShopBundle:Attribute')->findByLangAndShop(
    (int) $language['id_lang'], 
    (int) $language['shop_id']
);
```

@Quetzacoalt91 @matks wdyt?

Comment (dbadrukhiya) [reply src/PrestaShopBundle/Controller/Admin/ProductController.php]: Yes, I agree.

Comment (dineshbadrukhiya) [reply src/PrestaShopBundle/Controller/Admin/ProductController.php]: I think it should be like: 
`$language = empty($languages[0]) ? ['id_lang' => 1, 'shop_id' => 1] : $languages[0];
        $attributeGroups = $doctrine->getRepository('PrestaShopBundle:Attribute')->findByLangAndShop(
            (int) $language['id_lang'], 
            (int) ['shop_id']
        );`

Comment (PierreRambaud) [reply src/PrestaShopBundle/Controller/Admin/ProductController.php]: `(int) ['shop_id']` isn't an id..

**Review by dineshbadrukhiya — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10392)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @Mreker and @dbadrukhiya)

___
*[src/PrestaShopBundle/Controller/Admin/ProductController.php, line 546 at r1](https://reviewable.io/reviews/prestashop/prestashop/10392#-LMBjZ9MwGjg3UC4-FCt-r1-546:-LMI1n-Q6ptmT5WeXr2l:b-896fix) ([raw file](https://github.com/prestashop/prestashop/blob/f4d6d452326160f61f442341e22c0955b90ba6d5/src/PrestaShopBundle/Controller/Admin/ProductController.php#L546)):*
<details><summary><i>Previously, Mreker wrote…</i></summary><blockquote>

Should be:
`$attributeGroups = $doctrine->getRepository('PrestaShopBundle:Attribute')->findByLangAndShop((int) $languages[0]['id_lang'], (int) ['shop_id']);
`
</blockquote></details>

Done.
___
*[src/PrestaShopBundle/Controller/Admin/ProductController.php, line 546 at r2](https://reviewable.io/reviews/prestashop/prestashop/10392#-LMBjZ9MwGjg3UC4-FCt-r2-546:-LMM3DIh38epWYxBJoLE:b-ddygy2) ([raw file](https://github.com/prestashop/prestashop/blob/5106659459de77793a2165f294fbf699de962715/src/PrestaShopBundle/Controller/Admin/ProductController.php#L546)):*
<details><summary><i>Previously, PierreRambaud (GoT) wrote…</i></summary><blockquote>

`(int) ['shop_id']` isn't an id..
</blockquote></details>

id_shop ?

**Review by dineshbadrukhiya — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10392)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @Mreker and @PierreRambaud)

___
*[src/PrestaShopBundle/Controller/Admin/ProductController.php, line 546 at r2](https://reviewable.io/reviews/prestashop/prestashop/10392#-LMBjZ9MwGjg3UC4-FCt-r2-546:-LMMOLCfDT8F_R4ATSTB:b-896fix) ([raw file](https://github.com/prestashop/prestashop/blob/5106659459de77793a2165f294fbf699de962715/src/PrestaShopBundle/Controller/Admin/ProductController.php#L546)):*
<details><summary><i>Previously, dbadrukhiya (Dinesh Badrukhiya) wrote…</i></summary><blockquote>

id_shop ?
</blockquote></details>

Done.

**Comment by PierreRambaud** [comment]
@Quetzacoalt91 wdyt?

**Review by dineshbadrukhiya — COMMENTED** [review]
anything need to change? @PierreRambaud @Mreker

**Comment by dbadrukhiya** [comment]
> 
> 
> @Quetzacoalt91 wdyt?

I'm newbie on GitHub. I don't know how pull request approval works.  Can you please give me advise on my first pull request? Thanks.

**Comment by dineshbadrukhiya** [comment]
Can anyone help me to remove this. I was trying to open new request but by mistake it showing here and accidentally deleted theme.css file also.
untitled

File: src/PrestaShopBundle/Controller/Admin/ProductController.php [hunk 3]
```diff
@@ -543,7 +543,8 @@ public function formAction($id, Request $request)
         }
 
         $doctrine = $this->getDoctrine()->getManager();
-        $attributeGroups = $doctrine->getRepository('PrestaShopBundle:Attribute')->findByLangAndShop(1, 1);
+        $language = empty($languages[0]) ? ['id_lang' => 1, 'id_shop' => 1] : $languages[0];
+        $attributeGroups = $doctrine->getRepository('PrestaShopBundle:Attribute')->findByLangAndShop((int) $language['id_lang'], (int) ['id_shop']);
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/ProductController.php:547]: `(int) ['id_shop']` is not and id.

**Comment by PierreRambaud** [comment]
You must read https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History this is a really good article about how to rewrite history, from rewriting a git commit, to undo a commit, or squashing commits together =)

**Comment by dineshbadrukhiya** [comment]
Thanks @PierreRambaud

**Pull request closed without merging**